### PR TITLE
feat: change JS target to ES2020 

### PR DESCRIPTION
**Pull request by AllanZhengYP**
#### Description of changes

This change also aligns tslib across the board to `^2.5.0`

#### Description of how you validated changes
Tested on following tools
* Next.js 11 & 13
* Vue >= 3.2.12 (vite & vue-cli)
* Webpack 5
* Vite 4.x, 3.x, 2.9
* Metro >= 0.70

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by AllanZhengYP** [comment]
Don't merge as integration test update is still pending

**Review by elorzafe — DISMISSED** [review]
Amazing bundle size reduction!!! 

Kudos @AllanZhengYP 🏅 🎖️ 💯

File: .github/workflows/push-integ-test.yml [hunk 1]
```diff
@@ -7,8 +7,7 @@ concurrency:
 
 on:
   push:
-    branches:
-      - next/main
+    branches: ['next/main', 'es2020']
```
Comment (jimblanc) [inline .github/workflows/push-integ-test.yml:10]: Nit: Remove before merging

File: packages/api-graphql/__tests__/AWSAppSyncRealTimeProvider.test.ts [hunk 2]
```diff
@@ -342,7 +336,7 @@ describe('AWSAppSyncRealTimeProvider', () => {
 					// Watching for raised exception to be caught and logged
 					expect(loggerSpy).toBeCalledWith(
 						'DEBUG',
-						'error on bound ',
+						expect.stringContaining('error on bound '),
```
Comment (jimblanc) [inline packages/api-graphql/__tests__/AWSAppSyncRealTimeProvider.test.ts:339]: Nit: Why was this change needed? Additional data getting included in string?

File: packages/api/package.json [hunk 3]
```diff
@@ -53,7 +53,7 @@
 	"dependencies": {
 		"@aws-amplify/api-graphql": "4.0.0",
 		"@aws-amplify/api-rest": "4.0.0",
-		"tslib": "^2.6.1"
+		"tslib": "^2.5.0"
```
Comment (jimblanc) [inline packages/api/package.json:56]: Thanks for aligning these!

File: packages/core/package.json [hunk 4]
```diff
@@ -76,55 +76,43 @@
...
-			"name": "Custom clients (retry middleware)",
-			"path": "./lib-esm/clients/middleware/retry/middleware.js",
-			"import": "{ retryMiddleware }",
-			"limit": "1.45 kB"
-		},
-		{
-			"name": "Custom clients (signing middleware)",
-			"path": "./lib-esm/clients/middleware/signing/middleware.js",
-			"import": "{ signingMiddleware }",
-			"limit": "6.81 kB"
```
Comment (jimblanc) [inline packages/core/package.json:109]: Why remove these middleware tests?

**Review by jimblanc — APPROVED** [review]
Great stuff!! Thanks for driving this, some nits & questions below but overall looks good

File: packages/api-graphql/__tests__/AWSAppSyncRealTimeProvider.test.ts (hunk 2 above)
Comment (AllanZhengYP) [reply packages/api-graphql/__tests__/AWSAppSyncRealTimeProvider.test.ts:339]: @jimblanc The full message is `'error on bound _initializeHandshake'`. It comes from the `function.prototype.name` value of https://github.com/aws-amplify/amplify-js/blob/af048463dfec201394920e93dd997475ab1a167c/packages/api-graphql/src/Providers/AWSAppSyncRealTimeProvider/index.ts#L753

According to [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name#bound_function) this is expected. The function name was not shown before I guess has something to do with the `class` keyword. Originally the function name is anonymous, so the function name is `''`. Now with class keyword, the actual function name gets displayed now.

File: packages/core/package.json (hunk 4 above)
Comment (AllanZhengYP) [reply packages/core/package.json:109]: Because these code paths are always covered by other size limit test

**Review by jimblanc — DISMISSED** [review]

**Comment by jimblanc** [comment]
Approving pending a clean integ test run

**Review by jimblanc — APPROVED** [review]

File: scripts/rollup-externals.js [hunk 5]
```diff
@@ -21,10 +21,7 @@ const api_externals = [
 	'@aws-amplify/core',
 ];
 
-const auth_externals = [
-	'@aws-amplify/core',
-	'amazon-cognito-auth-js',
-];
+const auth_externals = ['@aws-amplify/core', 'amazon-cognito-auth-js'];
```
Comment (elorzafe) [inline scripts/rollup-externals.js:24]: I am not sure we need that `amazon-cognito-auth-js` anymore

Comment (AllanZhengYP) [reply scripts/rollup-externals.js:24]: This config is used by build script. I will remove all the build script-related things as follow-up

**Review by elorzafe — APPROVED** [review]
AWESOME WORK @AllanZhengYP 🏅 🎉
